h3i/src/frame.rs

```diff
@@ -113,12 +109,15 @@ impl From<Vec<Header>> for H3iFrame {
     }
 }
 
+pub type HeaderMap = HashMap<Vec<u8>, Vec<Vec<u8>>>;
+
 /// An HTTP/3 HEADERS frame with decoded headers and a [HeaderMap].
 #[derive(Clone, PartialEq, Eq)]
 pub struct EnrichedHeaders {
     header_block: Vec<u8>,
     headers: Vec<Header>,
-    header_map: HeaderMap<Vec<u8>>,
+    /// A multi-map of raw header names to values, similar to http's HeaderMap.
+    header_map: HeaderMap,
```

Comment:
This isn't a multi-map from a writing perspective, since its just backed by a raw HashMap. Yes, the parsing code emulates some multimap like behaviour but anyone naively using this type will be surprised. Consider instead using https://docs.rs/multimap/0.10.0/multimap/

Reply:
Nice